### PR TITLE
chore: prepare for 1.2.0

### PR DESCRIPTION
**Pull request by henryiii**
Would a 1.2 release make sense?

I also single-sourced the version, unless there's some reason we weren't doing that.

Also, https://github.com/pypa/pyproject-metadata/issues/68#issuecomment-2020350973 has instructions for signing GitHub release with Sigstore, we could set that up if desired. PyPI no longer allows GPG signing, which is still in our instructions (it just gets stripped out, I tried it on the last release).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -31,7 +30,8 @@ urls.changelog = "https://build.pypa.io/en/stable/changelog.html"
 urls.homepage = "https://build.pypa.io"
 urls.issues = "https://github.com/pypa/build/issues"
 urls.source = "https://github.com/pypa/build"
-
+# Version read from __version__ field in __init__.py by Flit
+dynamic = ["version"]
```
Comment (layday) [inline pyproject.toml:34]: Is flit clever about this or will it require installing all of our runtime deps?

Comment (henryiii) [reply pyproject.toml:34]: I'm pretty sure it's clever about it. But good point, let me check.

**Review by gaborbernat — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
Comment (henryiii) [reply pyproject.toml:34]: https://github.com/pypa/flit/blob/8ead22d4d3028f2dd2c952c85e5bf1535cca3b40/flit_core/flit_core/common.py#L140

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,7 +6,20 @@ Changelog
 Unreleased
```
Comment (layday) [inline CHANGELOG.rst:6]: Rm?

Comment (henryiii) [reply CHANGELOG.rst:6]: I left that there for our next release, but yes, I can drop it until we have something to put there.

Comment (layday) [reply CHANGELOG.rst]: Only because it's gonna be visible in the published docs.

**Review by layday — APPROVED** [review]

**Pull request merged**